### ISSUE TITLE
Corner cases for fenced div attributes

### ISSUE THREAD

**Issue by Witiko** [jgm/lunamark#48]
In #36, we added support for [Pandoc's fenced divs][1]. However, the fenced div attributes are treated as infostrings and only matched at capture time, which has a number of repercussions:

1. Our implementation parses attributes as fenced code infostrings. Infostrings [may not contain backticks][2], whereas attributes may (in Pandoc):

    ``` md
    ::: {key=value`}
    foo bar
    :::
    ```

2. Divs with empty infostrings are not divs. Consider the following document:

    ``` md
    :::
    foo
    :::
    ```

    Pandoc converts this document to the following HTML output: `<p>::: foo :::</p>`, where as we still consider this a div.

 [1]: https://pandoc.org/MANUAL.html#extension-fenced_divs
 [2]: https://github.com/jgm/lunamark/blob/master/lunamark/reader/markdown.lua#L231-L232

**Comment by Witiko** [comment]
@Omikhleia ping

**Comment by Omikhleia** [comment]
As for (1), the backtick issue is noted in #46. I would consider it very minor -- I don't quite see the real-world use case for backtick in attribute values (frankly, how many people would need this? I'd bet most logical usages are in the line of `{ widfth=200 custom-style="Framed" }`...

As for (2), it could be interested to check why Pandoc explicitly forbid empty divs, at the time they designed that beast. I found discussions on the mailing lists, but it's a very long read and I failed so far to find a clean formal syntax definition (I suspect one could say the code is the syntax, lol, but my skills in reading Haskell are close to none)

**Comment by Witiko** [comment]
> As for (2), it could be interested to check why Pandoc explicitly forbid empty divs, at the time they designed that beast.

Without attributes, it would be impossible to differentiate between an opening tag and a closing tag in the following code:

``` md
:::
Fenced Divs can be nested

:::
This is a nested Div
:::
:::
```

With attributes, we can easily differentiate between opening and closing fences.

**Comment by Witiko** [comment]
Closed via #51.
